test/jdk/sun/java2d/cmm/ColorConvertOp/FilterSemiCustomImages.java

```diff
@@ -0,0 +1,161 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+import java.awt.Point;
+import java.awt.color.ColorSpace;
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorConvertOp;
+import java.awt.image.ColorModel;
+import java.awt.image.SampleModel;
+import java.awt.image.WritableRaster;
+import java.io.File;
+
+import javax.imageio.ImageIO;
+
+import static java.awt.image.BufferedImage.TYPE_3BYTE_BGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR_PRE;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB_PRE;
+import static java.awt.image.BufferedImage.TYPE_INT_BGR;
+import static java.awt.image.BufferedImage.TYPE_INT_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_GRAY;
+
+/**
+ * @test
+ * @bug 8366208
+ * @summary Verifies ColorConvertOp works correctly with BufferedImage and
+ *          semi-custom raster
+ */
+public final class FilterSemiCustomImages {
+
+    private static final int W = 144;
+    private static final int H = 123;
+
+    private static final int[] TYPES = {
+            TYPE_INT_RGB, TYPE_INT_ARGB, TYPE_INT_ARGB_PRE, TYPE_INT_BGR,
+            TYPE_3BYTE_BGR, TYPE_4BYTE_ABGR, TYPE_4BYTE_ABGR_PRE,
+            TYPE_USHORT_GRAY
+    };
+
+    private static final int[] CSS = {
+            ColorSpace.CS_CIEXYZ, ColorSpace.CS_GRAY, ColorSpace.CS_LINEAR_RGB,
+            ColorSpace.CS_PYCC, ColorSpace.CS_sRGB
+    };
+
+    private static final class CustomRaster extends WritableRaster {
+        CustomRaster(SampleModel sampleModel, Point origin) {
+            super(sampleModel, origin);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        for (int fromIndex : CSS) {
+            for (int toIndex : CSS) {
+                if (fromIndex != toIndex) {
+                    for (int type : TYPES) {
+                        test(fromIndex, toIndex, type);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void test(int fromIndex, int toIndex, int type)
+            throws Exception
+    {
```

Comment:
```suggestion
    private static void test(int fromIndex, int toIndex, int type)
            throws Exception {
```
Java code style prefers the opening brace is on the same line instead of its own line.

Reply:
the java2d code uses the style that the "{" is on new line if the line was wrapped. That was forced by Jim, and I still follow that.

Reply:
Okay. How much common is it for the newly added code? I haven't come across it frequently recently.